readme.md

```diff
@@ -44,7 +44,7 @@ Yields:
 
 ## API
 
-### visit([node](https://github.com/wooorm/unist#unist-nodes)\[, type\], callback\[, reverse\])
+### visit([node](https://github.com/wooorm/unist#unist-nodes)\[, type], callback\[, reverse])
```

Comment:
> A reference in `text`, without definition node, no need to escape (currently is);

This is what I meant by "unsustainable": if we remove escaping for references without  accompanying definitions, in the future when such a definition gets added to the document, these references in older parts of the document will become valid and rendered as such! I wouldn't want this fragile behavior from remark. (Related: `[foo]` is parsed into a reference node regardless of whether a definition for `foo` exists, isn't it?)

> A reference in `text`, a definition node, needs escaping (maybe only the definition node);

Well, this is quite an interesting idea, but frankly this doesn't look nice to create a reference node and only [rely on a compiler](https://github.com/wooorm/remark-html/blob/cd388f9b3878a4b608b4b4489a061fada558aa28/lib/compilers.js#L57-L62) to not render it as a reference (there may be a compiler that throws errors or warnings on broken references or renders them somehow differently). Also plugins and utilities that work with references would probably not work as expected.